sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -308,6 +342,26 @@ class WhileStatementExec(
             condition
           case WhileState.Body =>
             val retStmt = body.getTreeIterator.next()
+
+            // Handle LEAVE or ITERATE statement if it has been encountered.
+            retStmt match {
+              case leaveStatementExec: LeaveStatementExec if !leaveStatementExec.hasBeenMatched =>
+                if (label.contains(leaveStatementExec.label)) {
```

Comment:
Is there a reason why we use `contains` instead of `equals`?

Reply:
it's not `contains` like in Strings, it's from the perspective of `Option`, i.e. it means whether the value within the `Option` is `leaveStatementExec.label`